manifests/copr.pp

```diff
@@ -0,0 +1,79 @@
+# Define: yum::copr
+#
+# This definition manages Copr (Cool Other Package Repo) repositories.
+#
+# Parameters:
+#   [*copr_repo*] - name of repository, defaults to title
+#   [*ensure*]    - specifies if repo should be enabled,
+#                   disabled or removed
+#
+# Actions:
+#
+# Requires:
+#   RPM based system
+#
+# Sample usage:
+#   yum::copr { 'copart/restic':
+#     ensure  => 'enabled',
+#   }
+#
+define yum::copr (
+  String                                 $copr_repo = $title,
+  Enum['enabled', 'disabled', 'removed'] $ensure    = 'enabled',
+) {
+  $prereq_plugin = $facts['package_provider'] ? {
+    'dnf'   => 'dnf-plugins-core',
```

Comment:
I always prefer to make the future, dnf in this case, the default case since it's more obvious what code needs removing once yum is gone from the universe.

Reply:
That's a good idea, thanks for the suggestion! I've also addressed this in the last push.